src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/TransportRestoreSnapshotAction.java

```diff
@@ -65,7 +65,7 @@ protected RestoreSnapshotResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(RestoreSnapshotRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
This might be `WRITE` actually - we are creating an index or modifying an index after all.